CMSgov/design-system
# [WNMGDS-760] Convert DateField and ChoiceList components to typescript

**Please follow the format below and remove any sections that aren't relevant.**
## Summary
- Updated DateField and ChoiceList components to typescript
- There were issues in the DateField component because of defaultDateFormatter being exported and the typescript doc generator thinking that it needed docs generated. Pulled this out into a separate file to avoid conflicts
- The typescript doc generator also had issues with typescript classes that extend HTML props. At first it was printing all props, the components and the HTML props. I had to add special options to pass into the typescript doc generator so that it would only include the component props

## How to test
http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-760/convert-datefiled-choicelist-to-typescript
- Review doc pages for DateField and ChoiceList and compare to existing doc site. Make sure there are no mismatches
- Test DateField and ChoiceList in create-react-app and create-react-app-typescript and make sure behavior is as expected
